### PR TITLE
guides - clarifications

### PR DESCRIPTION
**Pull request by bmacnaughton**
- show valid trace-settings for transactionSettings
- reorganize instrumenting-a-module.md
- thanks jiri tomek.

### CODE REVIEW

**Comment by Katulus** [comment]
Did you just move "probes" section to the end or did you change anything in it as well?

The section "The span-info function's signature is span-info ()" in "https://github.com/appoptics/appoptics-apm-node/blob/jiri-guide-updates/guides/instrumenting-a-module.md" has incorrectly formatted descriptions of `name, kvpairs, finalize` under bullet list. It's all in one paragraph and hard to read.

**Runner functions** are not clear. What they are for? Is it the `fn` for `ao.instrument(span, fn, config, callback)`? It's not clear from the text.

In general I would add some full examples to the instrumentation section. to see all the pieces in single snippet.

**Comment by bmacnaughton** [comment]
> Did you just move "probes" section to the end or did you change anything in it as well?

Moving it was the big change as most people probably didn't want to start with instrumenting a module. It might even make sense to move that to a separate file

There were a few other changes
- it noted the options argument
- it added More Information; it's not attempting to be a comprehensive document for such a small audience.

**Comment by bmacnaughton** [comment]
> The section "The span-info function's signature is span-info ()" in "https://github.com/appoptics/appoptics-apm-node/blob/jiri-guide-updates/guides/instrumenting-a-module.md" has incorrectly formatted descriptions of name, kvpairs, finalize under bullet list. It's all in one paragraph and hard to read.

good catch - thanks.

**Comment by bmacnaughton** [comment]
> Runner functions are not clear. What they are for? Is it the fn for ao.instrument(span, fn, config, callback)? It's not clear from the text.

thanks for this feedback. they had always been called "runner" functions but are just the function-to-be-run. And they are described with no context or introduction.

**Comment by bmacnaughton** [comment]
> In general I would add some full examples to the instrumentation section. to see all the pieces in single snippet.

i agree with you but this will have to wait as a separate project. examples are a project by themselves but they will help someone trying to use this a lot.

File: guides/instrumenting-a-module.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ function takes four arguments:
 The signature is `ao.instrument (span, fn, config, callback)`.
 
 - `span` is either the string name of the span to be created or a function that returns the span to use.
-- `fn` is a function that will run the function to be instrumented.
+- `runner` is a function that will run the function to be instrumented.
```
Comment (Katulus) [inline guides/instrumenting-a-module.md]: The function signature still has `fn` so now this `runner` does not refer to anything.

Comment (bmacnaughton) [reply guides/instrumenting-a-module.md]: thanks - done.

File: guides/instrumenting-a-module.md [hunk 2]
```diff
@@ -68,10 +13,10 @@ Wherever possible, it is preferred that `ao.instrument(...)` is used, however
 there are some alternatives, which we'll explore later. The `ao.instrument(...)`
 function takes four arguments:
 
-The signature is `ao.instrument (span, fn, config, callback)`.
+The signature is `ao.instrument (span, runner, config, callback)`.
 
 - `span` is either the string name of the span to be created or a function that returns the span to use.
-- `fn` is a function that will run the function to be instrumented.
+- `runner` is a function that will run the function to be instrumented.
```
Comment (prettycoder) [inline guides/instrumenting-a-module.md]: maybe add `(see below)`?
"runner" is not as a clear a concept as span, config, or callback.

File: guides/instrumenting-a-module.md [hunk 3]
```diff
@@ -68,10 +13,10 @@ Wherever possible, it is preferred that `ao.instrument(...)` is used, however
 there are some alternatives, which we'll explore later. The `ao.instrument(...)`
 function takes four arguments:
 
-The signature is `ao.instrument (span, fn, config, callback)`.
+The signature is `ao.instrument (span, runner, config, callback)`.
 
 - `span` is either the string name of the span to be created or a function that returns the span to use.
```
Comment (prettycoder) [inline guides/instrumenting-a-module.md]: do you mean "a function that returns the name to use"?

File: guides/instrumenting-a-module.md (hunk 2 above)
Comment (bmacnaughton) [reply guides/instrumenting-a-module.md]: added

File: guides/instrumenting-a-module.md (hunk 3 above)
Comment (bmacnaughton) [reply guides/instrumenting-a-module.md]: i added (see below) where it is explained.

**Pull request merged**